### PR TITLE
Feature/361 vesting integration test

### PR DESCRIPTION
**Pull request by sadiabbasi**

### CODE REVIEW

File: src/cli/bootstrap/bootstrap.ts [hunk 1]
```diff
@@ -164,18 +169,28 @@ async function createNativeToken() {
...
     };
 
-    for (const [account, amount] of Object.entries(allocations)) {
-        await tokenContract.transfer(
-            'eosio.token',
+    await vestngContract.updatedate('2024-03-19T00:00:00', '2024-04-20T00:00:00', signer);
+
+    for (const [account, { id, amount }] of Object.entries(allocations)) {
+        await vestngContract.assignTokens(
+            'coinsale.tmy',
```
Comment (theblockstalk) [inline src/cli/bootstrap/bootstrap.ts:190]: this should be assigning from eosio.token still, which is where the tokens are

File: test/services/blockchain/contracts/VestingContract.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+        try {
+            await vestngContract.assignTokens('eosio.token', 'team.tmy', '3000.000000 LEOS', 1, signer);
+
+            await vestngContract.assignTokens('eosio.token', 'coinsale.tmy', '4000.000000 LEOS', 1, signer);
+
+            const trx1 = await vestngContract.withdraw(holder, signer);
+
+            expect(trx1.processed.receipt.status).toBe('executed');
+        } catch (e) {
+            console.log(e.message, JSON.stringify(e, null, 2));
```
Comment (sadiabbasi) [inline test/services/blockchain/contracts/VestingContract.integration.test.ts:94]: Internal Service Error {
        "code": 500,
        "error": {
          "code": 3050003,
          "name": "eosio_assert_message_exception",
          "what": "eosio_assert_message assertion failure",
          "details": [
            {
              "message": "assertion failure with message: must transfer positive quantity",
              "file": "cf_system.cpp",
              "line_number": 14,
              "method": "eosio_assert"
            }
          ]
        },

File: test/services/blockchain/contracts/VestingContract.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+        try {
+            await vestngContract.assignTokens('eosio.token', 'ops.tmy', '1000.000000 LEOS', 1, signer);
+            await vestngContract.assignTokens('eosio.token', 'ecosystm.tmy', '1000.000000 LEOS', 2, signer);
+
+            const trx1 = await vestngContract.withdraw(holder, signer);
+
+            expect(trx1.processed.receipt.status).toBe('executed');
+        } catch (e) {
+            console.log(e.message, JSON.stringify(e, null, 2));
```
Comment (sadiabbasi) [inline test/services/blockchain/contracts/VestingContract.integration.test.ts:111]: Internal Service Error {
        "code": 500,
        "error": {
          "code": 3050003,
          "name": "eosio_assert_message_exception",
          "what": "eosio_assert_message assertion failure",
          "details": [
            {
              "message": "assertion failure with message: must transfer positive quantity",
              "file": "cf_system.cpp",
              "line_number": 14,
              "method": "eosio_assert"
            }
          ]
        },

File: test/services/blockchain/contracts/VestingContract.integration.test.ts (hunk 2 above)
Comment (sadiabbasi) [reply test/services/blockchain/contracts/VestingContract.integration.test.ts:94]: i think  claimable calculation is -ve value that's why this error happen?

File: test/services/blockchain/contracts/VestingContract.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,250 @@
...
+        for (let i = 0; i < 2; i++) {
+            const { user } = await createRandomID();
+            const userAccountName = await user.storage.accountName;
+            const accountInfo = await getAccountInfo(userAccountName);
+            const accountName = accountInfo.account_name.toString();
+
+            await vestngContract.assignTokens('eosio.token', accountName, '500.000000 LEOS', category, signer);
+            const trx = await vestngContract.withdraw(accountName, signer);
+
+            expect(trx.processed.receipt.status).toBe('executed');
```
Comment (sadiabbasi) [inline test/services/blockchain/contracts/VestingContract.integration.test.ts:229]: missing authority error for the withdrawal function as we have require_auth(holder) in withdraw

**Pull request merged**